src/core/utils/helpers/lazy-load.ts

```diff
@@ -0,0 +1,29 @@
+import { useEffect, useRef, useState } from "react";
+import { useIntersection } from "react-use";
+
+export const useListItemLazyLoad = () => {
```

Comment:
Naming here is a bit odd to me. Based on the implementation I cannot tell what the hook has to do with list items or lazy loading.

To me it might as well be called `hasBeenVisible()`, return `{ ref, hasBeenVisible }` and perhaps even accept an `IntersectionObserverInit` object.

Another option might be to add a DocBlock explaining the purpose of the hook.

Reply:
Renamed it